lib/rules/no-lone-blocks.js

```diff
@@ -11,12 +11,46 @@
 
 module.exports = function(context) {
 
+    /**
+     * Checks if the block includes any block-level bindings
+     * @param {ASTNode} block - The block node to be checked.
+     * @param {boolean} isStrict - True if the block scope is in strict mode.
+     * @returns {boolean} True if the block includes block-level bindings.
+    */
+    function hasBlockBindings(block, isStrict) {
+        var statement,
+            i, l;
+
+        for (i = 0, l = block.body.length; i < l; ++i) {
```

Comment:
I wanted to warn only about truly unnecessary blocks. Even with `blockBindings: true` you could have something like:
```js
{
  var x = 1
}

// or:
{
  foo();
}
```

Could we maybe use Escope info to infer whether the current block introduces any bindings, instead of doing this traversal?

Otherwise we could go with a simpler version where don't check for the actual presence of block-level bindings.